test/functional/admin/preview_controller_test.rb

```diff
@@ -29,24 +29,20 @@ class Admin::PreviewControllerTest < ActionController::TestCase
   end
 
   view_test "renders attached files if attachment_ids provided" do
-    edition = create(:published_detailed_guide, :with_file_attachment, body: "!@1")
+    edition = create(:published_detailed_guide, :with_file_attachment, body: "#Heading\n\n!@1\n\n##Subheading")
 
     post :preview, params: { body: edition.body, attachment_ids: edition.attachments.map(&:id) }
-    assert_select ".document .body" do
-      assert_select_object edition.attachments.first
-    end
+
+    assert_select ".document .body .govspeak section div a[href=?]", edition.attachments.first.url
   end
 
   view_test "shows alternative_format_contact_email in attachment block if alternative_format_provider_id given" do
     edition = create(:published_detailed_guide, :with_file_attachment, body: "!@1")
     alternative_format_provider = create(:organisation, alternative_format_contact_email: "alternative@example.com")
 
     post :preview, params: { body: edition.body, attachment_ids: edition.attachments.map(&:id), alternative_format_provider_id: alternative_format_provider.id }
-    assert_select ".document .body" do
-      assert_select_object edition.attachments.first do
-        assert_select "a[href^=\"mailto:alternative@example.com\"]"
-      end
-    end
+
+    assert_select ".document .body .govspeak section div a[href=?]", edition.attachments.first.url
```

Comment:
Should this be expecting a mailto link, or does the attachment url fallback to a mailto if the alternative format contact email is provided?

Reply:
@ryanb-gds Good spot! 🕵🏻

I think you're right – it doesn't look like this test is covering what it claims to. I'll push up a fix!

Reply:
@ryanb-gds I've just changed that here:

https://github.com/alphagov/whitehall/compare/6631b7cb972122ce7d7f80166b8f6c5c97d23294..5f2ecfad6d0feb1361074e1c8745ccc41cf696e9